src/pyfmi/fmi3.pyx

```diff
@@ -675,6 +678,155 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
             raise FMUException(f"The variable {pyfmi_util.decode(variablename)} could not be found.")
         return FMIL3.fmi3_import_get_variable_base_type(variable)
 
+
+    #@enable_caching TODO: Doesnt work using enums with members with C-types, lets discuss
+    def get_model_variables(self,
+        type: Optional[FMI3_Type] = None,
+        include_alias: bool = True,
+        causality: Optional[FMI3_Causality] = None,
+        variability: Optional[FMI3_Variability] = None,
+        only_start: bool = False,
+        only_fixed: bool = False,
+        filter = None,
+        _as_list: bool = False) -> Dict[str, FMI3ModelVariable]:
```

Comment:
Can we remove `_as_list`, is only used in one place in `io.py` for FMI2, and can easily be adjusted.

Reply:
Fixed (will account for the removal in my other PR for result handling)